zebra-network/src/peer_set/initialize.rs

```diff
@@ -222,7 +257,9 @@ where
         if let Err((addr, ref e)) = handshake_result {
             info!(?addr, ?e, "an initial peer connection failed");
         }
-        tx.send(handshake_result.map_err(|(_addr, e)| e)).await?;
+        peer_set_sender
+            .send(handshake_result.map_err(|(_addr, e)| e))
+            .await?;
```

Comment:
Simplify this code by continuing on errors, and only sending on success - the peer set ignores errors anyway.